src/vsCodeConfig/settings.ts

```diff
@@ -43,6 +43,25 @@ export function getWorkspaceSetting<T>(key: string, fsPath?: string, prefix: str
     return projectConfiguration.get<T>(key);
 }
 
+/**
+ * Searches through all open folders and gets the current workspace setting (as long as there are no conflicts)
+ */
+export function getWorkspaceSettingFromAnyFolder(key: string, prefix: string = extensionPrefix): string | undefined {
+    // tslint:disable-next-line: strict-boolean-expressions
+    const folders: WorkspaceFolder[] = workspace.workspaceFolders || [];
+    let result: string | undefined;
+    for (const folder of folders) {
+        const projectConfiguration: WorkspaceConfiguration = workspace.getConfiguration(prefix, folder.uri);
+        const folderResult: string | undefined = projectConfiguration.get<string>(key);
+        if (!result) {
+            result = folderResult;
+        } else if (folderResult && result !== folderResult) {
```

Comment:
So if you have two different results, you ignore both of them?

Reply:
There is no root - they're all at the same level. In most cases returning undefined means the user will be prompted for the value, which I think makes complete sense